src/core/util/log.cc

```diff
@@ -123,6 +124,33 @@ static gpr_atm parse_log_severity(absl::string_view str, gpr_atm error_value) {
   return error_value;
 }
 
+void disable_vlog_for_grpc() { absl::SetVLogLevel("*grpc*/*", -1); }
+
+void gpr_to_absl_verbosity_setting_init(void) {
+  absl::string_view verbosity = grpc_core::ConfigVars::Get().Verbosity();
+  DVLOG(2) << "Log verbosity: " << verbosity;
+  if (absl::EqualsIgnoreCase(verbosity, "INFO")) {
+    LOG(WARNING) << "Not suitable for production. Prefer WARNING or ERROR. "
+                    "However if you see this message in a debug environmenmt "
+                    "or test environmenmt it is safe to ignore this message.";
+    absl::SetMinLogLevel(absl::LogSeverityAtLeast::kInfo);
+  } else if (absl::EqualsIgnoreCase(verbosity, "DEBUG")) {
+    LOG(ERROR) << "Not suitable for production. Prefer WARNING or ERROR. "
+                  "However if you see this message in a debug environmenmt or "
+                  "test environmenmt it is safe to ignore this message.";
+    absl::SetVLogLevel("*grpc*/*", 2);
+    absl::SetMinLogLevel(absl::LogSeverityAtLeast::kInfo);
+  } else if (absl::EqualsIgnoreCase(verbosity, "ERROR")) {
+    absl::SetMinLogLevel(absl::LogSeverityAtLeast::kError);
+  } else if (absl::EqualsIgnoreCase(verbosity, "NONE")) {
+    absl::SetMinLogLevel(absl::LogSeverityAtLeast::kInfinity);
+  } else if (absl::EqualsIgnoreCase(verbosity, "UNKNOWN")) {
```

Comment:
How does gpr_log handle the verbosity level "UNKNOWN"? Whats the behavior if we don't set to any level? Logging gets disabled?

Reply:
This log level will never be passed to gpr_log.
This will happen only and only when no value is present for GRPC_VERBOSITY. This would be the default value.